src/cloudflare/internal/sockets.d.ts

```diff
@@ -30,3 +32,5 @@ export function connect(
   address: string | SocketAddress,
   options?: SocketOptions
 ): Socket;
+
+export function newHttpClient(socket: Socket): Promise<Fetcher>;
```

Comment:
We likely want to stress in the documentation that `newHttpClient` is not a part of the "standard" `Socket` API that we're proposing in WinterTC.

Reply:
There's not going to be any documentation, this is an internal only experimental feature.

Reply:
Since it's internal-only, should we give it a name that emphasises this? `internalNewHttpClient` or something?

Reply:
That's what the experimental guard is for.